From 474e893fd122db6093d022adc6c37047f6a09f0c Mon Sep 17 00:00:00 2001
From: Zhou Qiankang <wszqkzqk@qq.com>
Date: Tue, 17 Jun 2025 15:06:54 +0800
Subject: [PATCH] HIP: Add __LP64__ to 64-bit platform detection

The HIP device pointer type was defined incorrectly on other 64-bit platforms
such as riscv64 and loongarcg64. This change ensures that `hipDeviceptr_t` is
defined as `unsigned long long` on these platforms.

Signed-off-by: Zhou Qiankang <wszqkzqk@qq.com>
---
 extern/hipew/include/hipew.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/extern/hipew/include/hipew.h b/extern/hipew/include/hipew.h
index e87daaec213..963c5092f0b 100644
--- a/extern/hipew/include/hipew.h
+++ b/extern/hipew/include/hipew.h
@@ -87,7 +87,7 @@ typedef uint32_t hipuint32_t;
 typedef uint64_t hipuint64_t;
 #endif
 
-#if defined(__x86_64) || defined(AMD64) || defined(_M_AMD64) || defined (__aarch64__) || defined(_M_ARM64) || defined(__ppc64__) || defined(__PPC64__)
+#if defined(__x86_64) || defined(AMD64) || defined(_M_AMD64) || defined (__aarch64__) || defined(_M_ARM64) || defined(__ppc64__) || defined(__PPC64__) || defined(__LP64__)
 typedef unsigned long long hipDeviceptr_t;
 #else
 typedef unsigned int hipDeviceptr_t;
